src/core/components/SubmitButton.tsx

```diff
@@ -0,0 +1,61 @@
+import { Button, ButtonTypeMap, CircularProgress, ExtendButtonBase } from '@mui/material';
```

Comment:
this core component is still a WIP. Its meant to be an analog/replacement for our ember `<ButtonSpinner>` and was extracted from Austin's code in the PKA app, which will show a loading state while waiting for the onClick callback promise to resolve, "shake" when the onClick() promise rejects, and (eventually) show a success checkmark icon flip-in animation when the promise resolves.